### ISSUE TITLE
坐标位置旋转代码是否有错误

### ISSUE THREAD

**Issue by yehx6** [jingyaogong/minimind#532]
model_minimind.py的代码中
def apply_rotary_pos_emb(q, k, cos, sin, position_ids=None, unsqueeze_dim=1):
    def rotate_half(x):
        return torch.cat((-x[..., x.shape[-1] // 2:], x[..., : x.shape[-1] // 2]), dim=-1)

    q_embed = (q * cos.unsqueeze(unsqueeze_dim)) + (rotate_half(q) * sin.unsqueeze(unsqueeze_dim))
    k_embed = (k * cos.unsqueeze(unsqueeze_dim)) + (rotate_half(k) * sin.unsqueeze(unsqueeze_dim))
    return q_embed, k_embed
这里是否应该是将奇数index的元素加符号，而不是向量后半部分加负号？

**Comment by jingyaogong** [comment]
没有错，你可以参考https://github.com/jingyaogong/minimind/issues/449
